x/gamm/keeper/pool_service.go

```diff
@@ -189,12 +192,13 @@ func (k Keeper) JoinPoolNoSwap(
 		return err
 	}
 
+	// we do an abstract calculation on the lp liquidity coins needed to have the designated amount of given shares of the pool
 	neededLpLiquidity, err := getMaximalNoSwapLPAmount(ctx, pool, shareOutAmount)
 	if err != nil {
 		return err
 	}
 
-	// if neededLPLiquidity >= tokenInMaxs, return err
+	// check that needed lp liquidity does not exceed `tokenInMaxs`. Return error if so.
```

Comment:
I think it makes it clearer that this parameter is provided by the caller

```suggestion
	// check that needed lp liquidity does not exceed the given `tokenInMaxs` parameter. Return error if so.
```